### PR TITLE
Add named registration support for DI container & avoid recreating object on init

### PR DESCRIPTION
**Pull request by vishal-kr-barnwal**
Updated DI functions to support optional name parameters for better instance management and retrieval. Improved initialization logic in `GetItDIPlugin` by directly resetting the DI Instance.

### CODE REVIEW

File: packages/system/vyuh_core/lib/plugin/di/di_plugin.dart [hunk 1]
```diff
@@ -10,23 +10,23 @@ abstract class DIPlugin extends Plugin {
...
-  void unregister<T extends Object>();
+  void unregister<T extends Object>({String? name});
 
   /// Retrieves the registered instance of the specified Object type from the DI container.
-  T get<T extends Object>();
+  T get<T extends Object>({String? name});
 
   /// Checks if the instance of the specified Object type is registered in the DI container.
-  bool has<T extends Object>();
+  bool has<T extends Object>({String? name});
```
Comment (pavanpodila) [inline packages/system/vyuh_core/lib/plugin/di/di_plugin.dart:29]: the di plugin should also use the name ... not being used right now

**Review by pavanpodila — CHANGES_REQUESTED** [review]
use the name in getit di

File: packages/system/vyuh_core/lib/plugin/di/di_plugin.dart (hunk 1 above)
Comment (vishal-kr-barnwal) [reply packages/system/vyuh_core/lib/plugin/di/di_plugin.dart:29]: In Get_it is already being used. Exposed this as a feature enhancement where feature builders can register named types.

**Comment by vishal-kr-barnwal** [comment]
name is already being used in get_it plugin thus didn't made any change there. it just that there was no way to access named DI without TypeCasting which is an extra step for feature development.

**Review by pavanpodila — APPROVED** [review]

**Pull request merged**